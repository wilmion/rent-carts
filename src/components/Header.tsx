import React from 'react';
import { useHistory } from 'react-router';
import { connect } from 'react-redux';

import { AiOutlineMenu , AiFillCloseCircle } from 'react-icons/ai';
import { FaUser } from 'react-icons/fa';
import { BiArrowBack } from 'react-icons/bi';

import { IState , IUser } from '../models/interface';

import "../sass/layout/header.scss"


const Header:React.FC<{user:IUser | null}> = (props) => {
    
    const history = useHistory();

    const navigateLink = (url:string):void => {
        const menu:HTMLElement = document.querySelector('.header-menu');

        menu.style.transform = 'translateX(-100%)';

        history.push(url);
    }

    const showMenu = (position:number):void => {
        const menu:HTMLElement = document.querySelector('.header-menu');
        menu.style.transform = `translateX(${position}%)`;
    }

    return (
        <header className="header">
            <section className="header-menu" >
                <div className="header-menu-close">
                    <AiFillCloseCircle onClick={() => showMenu(-100)} className="header-menu-close__icon" />
                </div>
                <p className="header-menu__section" onClick={() => navigateLink( '/' )} >Home</p>
                <p className="header-menu__section" onClick={() => navigateLink( '/profile' )} >{props.user? "My Acount" :"Log In"}</p>
                <p className="header-menu__section" onClick={() => navigateLink( '/carts' )} >Type Cars</p>
                <p className="header-menu__section" onClick={() => navigateLink( '/about ' )} >About</p>
            </section>
            <section className="header-show" id="header">
                <BiArrowBack className="header-show__icon-left" size="20px" onClick={() => history.goBack()} ></BiArrowBack>
                <h3 className="header-show__title">Find your car</h3>
                <nav className="header-show-nav">
                    <div className="header-show-nav-logo">
                        <img onClick={() => navigateLink('/')} src="https://firebasestorage.googleapis.com/v0/b/rentcarts.appspot.com/o/logo.png?alt=media&token=5392288a-fd79-4cfe-a99e-f4306ccedd88" alt="Favicon"/>
                    </div>
                    <ul className="header-show-nav-list">
                        <li className="header-show-nav-list__item" onClick={() => navigateLink('/')} >Home</li>
                        <li className="header-show-nav-list__item" onClick={() => navigateLink('/carts')} >Cars</li>
                        <li className="header-show-nav-list__item" onClick={() => navigateLink('/about')} >About</li>
                        <li className="header-show-nav-list__item">|</li>
                        {props.user? (
                        <>
                            <li className="header-show-nav-list__item" ><FaUser className="header-show-nav-list__item--icon" /></li>
                            <li className="header-show-nav-list__item" onClick={() => navigateLink( '/profile' )} >{props.user.username}</li>
                        </>
                        ): (
                            <li className="header-show-nav-list__item"><button onClick={() => navigateLink( '/login' )} className="header-show-nav-list__item--button">Log In</button></li>
                        )}
                        
                    </ul>
                </nav>
                <div className="header-show-menu">
                    <AiOutlineMenu size="20px" className="header-show-menu-icon" onClick={() => showMenu(0)} ></AiOutlineMenu>
                </div>
            </section>
        </header> 
    )
}

const mapStateToProps = (state:IState) => ({
    user: state.user
})

export default connect(mapStateToProps , null)(Header)
