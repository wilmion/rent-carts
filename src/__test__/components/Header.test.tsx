import React from 'react';
import { mount , shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/providerMock';
import Header from '../../components/Header';

describe('<Header />' , () => {
    test('Render Component' , () => {
        const header = shallow(<ProviderMock><Header /></ProviderMock>);
        expect(header.length).toEqual(1);
    })
    test('Snapshot' , () => {
        const headerJSON = create(<ProviderMock><Header /></ProviderMock>).toJSON();
        expect(headerJSON).toMatchSnapshot();
    })
})