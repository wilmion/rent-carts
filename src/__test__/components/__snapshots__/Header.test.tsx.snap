// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot 1`] = `
<header
  className="header"
>
  <section
    className="header-menu"
  >
    <div
      className="header-menu-close"
    >
      <svg
        className="header-menu-close__icon"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </div>
    <p
      className="header-menu__section"
      onClick={[Function]}
    >
      Home
    </p>
    <p
      className="header-menu__section"
      onClick={[Function]}
    >
      Log In
    </p>
    <p
      className="header-menu__section"
      onClick={[Function]}
    >
      Type Cars
    </p>
    <p
      className="header-menu__section"
      onClick={[Function]}
    >
      About
    </p>
  </section>
  <section
    className="header-show"
    id="header"
  >
    <svg
      className="header-show__icon-left"
      fill="currentColor"
      height="20px"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="20px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 11L6.414 11 11.707 5.707 10.293 4.293 2.586 12 10.293 19.707 11.707 18.293 6.414 13 21 13z"
      />
    </svg>
    <h3
      className="header-show__title"
    >
      Find your car
    </h3>
    <nav
      className="header-show-nav"
    >
      <div
        className="header-show-nav-logo"
      >
        <img
          alt="Favicon"
          onClick={[Function]}
          src="https://firebasestorage.googleapis.com/v0/b/rentcarts.appspot.com/o/logo.png?alt=media&token=5392288a-fd79-4cfe-a99e-f4306ccedd88"
        />
      </div>
      <ul
        className="header-show-nav-list"
      >
        <li
          className="header-show-nav-list__item"
          onClick={[Function]}
        >
          Home
        </li>
        <li
          className="header-show-nav-list__item"
          onClick={[Function]}
        >
          Cars
        </li>
        <li
          className="header-show-nav-list__item"
          onClick={[Function]}
        >
          About
        </li>
        <li
          className="header-show-nav-list__item"
        >
          |
        </li>
        <li
          className="header-show-nav-list__item"
        >
          <button
            className="header-show-nav-list__item--button"
            onClick={[Function]}
          >
            Log In
          </button>
        </li>
      </ul>
    </nav>
    <div
      className="header-show-menu"
    >
      <svg
        className="header-show-menu-icon"
        fill="currentColor"
        height="20px"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </div>
  </section>
</header>
`;
